Sources/MessagingPush/CustomerIOParsedPushPayload.swift

```diff
@@ -103,11 +103,17 @@ public class CustomerIOParsedPushPayload {
     public let mutableNotificationContent: UNMutableNotificationContent
     public let notificationContent: UNNotificationContent
 
+    // We are trying to isolate the SDK from the UserNotifications framework entirely. So, it could be good to decouple this class, too.
+    // Until then, this parse() function is added to take the abstract PushNotification data type and call one of the UserNotification parse() functions.
```

Comment:
De-coupling this file from the `UserNotifications` framework might be out-of-scope for the current task. As we continue writing tests for the code in the push module, this `parse()` function may be the only `parse()` function in the file and the `pushNotification as? UNNotificationWrapper` line will not be needed.

Reply:
Thanks for adding this context because i was wondering the same thing, why do we care if its `pushNotification is UNNotification` to proceed, ideally since `PushNotification` is the only source of truth, no matter who is creating it, we should proceed with it as it would make writing the test cases easier.

Reply:
> ideally since PushNotification is the only source of truth, no matter who is creating it, we should proceed with it as it would make writing the test cases easier.

You are spot on with this comment. 

Glad the comment was helpful!